highlight/HighlightsFromPointsExplainer.md

```diff
@@ -0,0 +1,60 @@
+ - When a user hovers over a misspelled word, the web app may display UI with suggested replacement text.
+ - When a user clicks an annotation in a document, the web app may emphasize and scroll into view the corresponding annotation in a pane which lists all the annotations in the document.
+
+
+## Proposal
+The highlightsFromPoint() API is designed to enhance user interaction with highlighted content on web pages. Highlights are used to visually distinguish specific ranges within a document, such as comments, search results, or spelling errors. This API aims to provide a robust mechanism for identifying and interacting with these highlights, addressing challenges related to multiple overlapping highlights, event coordination, shadow DOM handling, and performance optimization.
+
+### Key Considerations 
+- **Multiple Overlapping Highlights**: When multiple highlights overlap (e.g., from a spell-checker and a find-on-page feature), it's crucial to identify all highlights at a specific point. This allows web developers to determine which highlight should receive the event first, ensuring the correct highlight responds to user interactions.
+- **Event Coordination**: Different features might create highlights independently. This API provides a way to coordinate events between these highlights, ensuring each highlight can handle events appropriately without conflicts.
+- **Shadow DOM Handling**: Highlights within shadow DOMs require special handling to maintain encapsulation. The method can be designed to respect shadow DOM boundaries, ensuring highlights inside shadows are managed correctly. This helps maintain the integrity of the shadow DOM while still allowing highlights to be identified and interacted with.
+- **Performance Optimization**: By providing a dedicated API for hit-testing highlights, the method can optimize performance. Instead of relying on more complex and potentially slower methods to determine which highlights are under a specific point, this method offers a streamlined and efficient way to perform this task, improving overall performance.
+
+### API Design
+The CSS.highlights.highlightsFromPoint() method will be part of the CSS.highlights interface. It will return a sequence of highlights at a specified point, ordered by [priority](https://drafts.csswg.org/css-highlight-api-1/#priorities). The developer has the option to pass in `options`, which is an optional dictionary where the key maps to an array of `ShadowRoot` objects. The API can search for and return highlights within the provided shadow DOM. The approach of passing an `optiona` parameter of `ShadowRoom` object is similar to [caretPositionFromPoint()](https://drafts.csswg.org/cssom-view/#dom-document-caretpositionfrompoint) and [getHTML()](https://html.spec.whatwg.org/multipage/dynamic-markup-insertion.html#dom-element-gethtml) methods. 

```

Comment:
```suggestion
The `highlightsFromPoint()` method will be part of the `CSS.highlights` interface. It will return a sequence of highlights at a specified point, ordered by [priority](https://drafts.csswg.org/css-highlight-api-1/#priorities). The developer has the option to pass in `options`, which is an optional dictionary where the key maps to an array of `ShadowRoot` objects. The API can search for and return highlights within the provided shadow DOM. The approach of passing an `options` parameter of `ShadowRoot` object is similar to [caretPositionFromPoint()](https://drafts.csswg.org/cssom-view/#dom-document-caretpositionfrompoint) and [getHTML()](https://html.spec.whatwg.org/multipage/dynamic-markup-insertion.html#dom-element-gethtml) methods. 
```